papaya-web/src/main/java/pl/edu/agh/papaya/rest/sprints/controller/SprintsController.java

```diff
@@ -0,0 +1,49 @@
+package pl.edu.agh.papaya.rest.sprints.controller;
+
+import com.google.common.collect.Lists;
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import pl.edu.agh.papaya.api.model.SprintStateDto;
+import pl.edu.agh.papaya.api.model.SprintWithStateDto;
+import pl.edu.agh.papaya.api.service.SprintsApi;
+import pl.edu.agh.papaya.mappers.SprintMapper;
+import pl.edu.agh.papaya.mappers.SprintStateMapper;
+import pl.edu.agh.papaya.model.SprintState;
+import pl.edu.agh.papaya.service.sprint.SprintService;
+
+@RestController
+public class SprintsController implements SprintsApi {
+
+    @Autowired
+    private SprintService sprintService;
+
+    @Autowired
+    private SprintStateMapper sprintStateMapper;
+
+    @Autowired
+    private SprintMapper sprintMapper;
+
+    @Override
+    public ResponseEntity<List<SprintWithStateDto>> getSprints(List<SprintStateDto> sprintStateDtos) {
+        final LocalDateTime currentTime = LocalDateTime.now();
+
+        List<SprintState> sprintStates = sprintStateMapper.mapFromApi(
+                Optional.ofNullable(sprintStateDtos).orElseGet(() -> Lists.newArrayList(SprintStateDto.values())));
```

Comment:
nie wiem, czy to jakoś uprości (albo czy ma sens), ale może mapper mógłby sobie trzymać już zmapowane SprintStateDto.values(), dostępne pod jakąś metodą?

Reply:
Uszczęśliwi Cię pole `final static`?

Reply:
będzie piękne <3
a static bym opuścił, skoro i tak zawsze jest dostępny obiekt (no ale nie zmniejsza to poziomu uszczęśliwienia)